packages/core/src/types.ts

```diff
@@ -5,6 +5,8 @@ import { ReaderTaskEither } from 'fp-ts/ReaderTaskEither';
 import { TaskEither } from 'fp-ts/TaskEither';
 import { Logger } from 'pino';
 import { NonEmptyArray } from 'fp-ts/NonEmptyArray';
+import { URL } from 'url';
```

Comment:
This is not required, are you receiving any error?

Reply:
You're right, it's not required. I guess the issue I was running into was that without the URL type in the module typescript seems to think that `upstream` is an `any` type and was not warning me when I passed in a string as a url.  Maybe this is an issue with my tsconfig or vscode?

<img width="274" alt="Screen Shot 2020-09-09 at 4 55 52 PM" src="https://user-images.githubusercontent.com/8688257/92663125-a63d5700-f2bd-11ea-9c24-6770664dd0a4.png">
<img width="514" alt="Screen Shot 2020-09-09 at 4 57 08 PM" src="https://user-images.githubusercontent.com/8688257/92663127-a6d5ed80-f2bd-11ea-8226-58bcab599521.png">

Reply:
I'll take a look into. Can you remote the import in meantime?

Reply:
I've done that for you. This is good to merge.